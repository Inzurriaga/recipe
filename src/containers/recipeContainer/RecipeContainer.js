import React, { Component } from "react"
import { connect } from "react-redux"
import RecipeCard from "../recipeCard/RecipeCard"

export class RecipeContainer extends Component {

  checkForRecipes = () => {
    if(this.props.recipes.meals){
        return this.props.recipes.meals.map((meal, index) => {
        let svgClass;
        if(index % 2 === 0){
          svgClass = "folder-tap-odd"
        }else{
          svgClass = "folder-tap-even"
        }
        return <RecipeCard key={meal.idMeal} info={meal} svgClass={svgClass}/>
      })
    }else{
      return(
        <div>
          <h2>no recipe found</h2>
        </div>
      )
    }
  }
  render(){
    const recipes = this.checkForRecipes()
    return(
      <section className="recipe-container">
        {
          recipes
        }
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  recipes: state.recipes
})

export default connect(mapStateToProps)(RecipeContainer)