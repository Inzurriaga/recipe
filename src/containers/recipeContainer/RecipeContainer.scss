.recipe-container {
  position: relative;
  overflow: scroll;
  display: flex;
  flex: 1fr;
  justify-content: center;
  top: 125px;
}