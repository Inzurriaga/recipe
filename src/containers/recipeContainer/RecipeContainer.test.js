import React from 'react';
import { RecipeContainer } from './RecipeContainer';
import { shallow } from "enzyme";

const mockRecipes = [];

describe("RecipeContainer", () => {

  describe("RecipeContainer", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <RecipeContainer recipes={mockRecipes}/>
      );
    })

    it("should match snap shot", () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

  describe("mapStateToProps", () => {

  })
})