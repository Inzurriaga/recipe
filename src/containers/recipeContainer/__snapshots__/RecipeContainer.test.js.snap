// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeContainer RecipeContainer should match snap shot 1`] = `ShallowWrapper {}`;
